### PR TITLE
feat: Add blogpost for 1.12 release

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Fix https://github.com/kubewarden/kubewarden.io/issues/220

### CODE REVIEW

**Review by kravciak — APPROVED** [review]

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+thanks to `spec.contextAwareResources`, has an increased granularity compared
+to vanilla Gatekeeper.
+
+The cache works with sets of resources as defined in the policy rules and
+`spec.contextAwareResources`. Hence policies that access the same set of
+resources (e.g: all Namespaces with a specific LabelSelector) will share the
+same inventory. That cache set is invalidated whenever a resource of that set
+changes.
+
+Stay tuned for more performance improvements on next releases!
```
Comment (flavio) [inline content/blog/2024/04/kubewarden-1-12-release.md:36]: It's a bit too low level.

```suggestion
The previous [1.11 release](https://www.kubewarden.io/blog/2024/03/kubewarden-1-11-release/) featured lots of optimizations for
[context aware policies](https://docs.kubewarden.io/next/reference/spec/context-aware-policies).

The 1.12 release provides a further optimization for Gatekeeper policies that access Kubernetes resources.
This optimization provides an extra 55% performance boost for these policies.
```

**Review by jhkrug — CHANGES_REQUESTED** [review]
Apart from some minor suggestions, LGTM.

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+This release focuses on optimizations and High-Availability, both oriented to
+production.
+
+## Optimizing Rego policies
+
+On really big clusters (thousands of nodes), Gatekeeper policies that make use
+of context-aware data are markedly slower than those written in Go/Rust. This
+is because Gatekeeper policies expect a serialized inventory of all Kubernetes
+objects they could access under `data.inventory`, for which the policy-server
+must spent significant CPU cycles.
```
[the remark this answers is not in the document]
Comment (viccuad) [reply content/blog/2024/04/kubewarden-1-12-release.md]: This block changed; I have applied some of its suggestions by hand.

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+Gatekeeper policies in Kubewarden in addition have
+`spec.contextAwareResources`, a Kubewarden feature that allows us to fine-tune
+the context-aware permissions per policy, and provide more security.
+
+Following requests from the community, for Kubewarden 1.12 we have elected to
+cache this `data.inventory` of Gatekeeper policies for performance. The cache,
+thanks to `spec.contextAwareResources`, has an increased granularity compared
+to vanilla Gatekeeper.
```
[the remark this answers is not in the document]
Comment (viccuad) [reply content/blog/2024/04/kubewarden-1-12-release.md]: This block changed; I have applied some of its suggestions by hand.

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+The cache works with sets of resources as defined in the policy rules and
+`spec.contextAwareResources`. Hence policies that access the same set of
+resources (e.g: all Namespaces with a specific LabelSelector) will share the
+same inventory. That cache set is invalidated whenever a resource of that set
+changes.
```
[the remark this answers is not in the document]
Comment (viccuad) [reply content/blog/2024/04/kubewarden-1-12-release.md]: This block changed; I have applied some of its suggestions by hand.

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+On really big clusters (thousands of nodes), Gatekeeper policies that make use
+of context-aware data are markedly slower than those written in Go/Rust. This
+is because Gatekeeper policies expect a serialized inventory of all Kubernetes
+objects they could access under `data.inventory`, for which the policy-server
+must spent significant CPU cycles.
+
+Gatekeeper policies in Kubewarden in addition have
+`spec.contextAwareResources`, a Kubewarden feature that allows us to fine-tune
+the context-aware permissions per policy, and provide more security.
```
[the remark this answers is not in the document]
Comment (viccuad) [reply content/blog/2024/04/kubewarden-1-12-release.md]: This block changed; I have applied some of its suggestions by hand.

**Review by jhkrug — APPROVED** [review]
lgtm

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+featured lots of optimizations for
+[context aware policies](https://docs.kubewarden.io/next/reference/spec/context-aware-policies).
+
+The 1.12 release provides a further optimization for Gatekeeper policies that
+access Kubernetes resources. This optimization provides an extra 55%
+performance boost for these policies.
+
+This is particularly noticeable on really big clusters (thousands of nodes), as
+Gatekeeper policies expect a serialized inventory of all Kubernetes objects
+they could access, under `data.inventory`.
```
Comment (flavio) [inline content/blog/2024/04/kubewarden-1-12-release.md:26]:
```suggestion
The benefits of this optimization are particularly noticeable when a huge number
of Kubernetes resources are accessed by a Gatekeeper policy.

For example, a policy that access Pod resources, will see a reduced evaluation time
on big clusters where hundreds of Pod objects are defined.
```

File: content/blog/2024/04/kubewarden-1-12-release.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+In addition, Gatekeeper policies in Kubewarden have `spec.contextAwareResources`
+which allows us to fine-tune the context-aware permissions per policy.
+Leveraging this feature allows us to provide greater cache granularity for each
+resource set shared between policies (e.g: "all Namespaces with a specific
+LabelSelector").
```
Comment (flavio) [inline content/blog/2024/04/kubewarden-1-12-release.md:32]:
```suggestion
```

I would drop this section. Gatekeeper doesn't use label selectors when accessing resources, hence we cannot filter that.

What we have always been doing is to expose to the Gatekeeper policy only the resources it's interested about.

With Gatekeeper the admin determines what has to be shared with policies, then this is provided to **all** policies, even if they do not care.

**Review by flavio — CHANGES_REQUESTED** [review]

**Review by jvanz — APPROVED** [review]

**Pull request merged**